modules/util/source-formatter/src/main/java/com/liferay/source/formatter/check/BaseSourceCheck.java

```diff
@@ -80,6 +80,23 @@ public int getWeight() {
 		return _weight;
 	}
 
+	public boolean hasParameterTypes(
+		String content, List<String> parameterList, String[] parameterTypes) {
+
+		for (int i = 0; i < parameterTypes.length; i++) {
+			String variableTypeName = getVariableTypeName(
+				content, content, parameterList.get(i), true);
+
+			if ((variableTypeName == null) ||
+				!parameterTypes[i].equals(variableTypeName)) {
+
+				return false;
+			}
+		}
+
+		return true;
+	}
+
```

Comment:
Hi @TamyrisTorres, I know we discussed this earlier.

I have some concerns about this particular automation/rule and the `hasParameterTypes` method. Checking the types of the arguments in method call is very difficult and error-prone when we are dealing with strings, especially when we consider literal values and method calls within method calls. Even with a dedicated Java parser, it is still a little tricky based on the SF framework's current capabilities.

I question whether this should be automated. At best, only the check should only handle the case when all the method arguments are variables. If the arguments are not all variables, then we ignore it. Let me know if that makes sense.